File: Custom_responses_that_don’t_hit_origin_servers/lua/prevent_a_specific_ip.lua
local _M = {}

local BLOCK_REG = [[1\.2\.3\.\d+]]

function _M.handler(event)
    local request = event.request

    -- 如果 client ip 属于 1.2.3.0/24
    -- 返回 403
    if re.find(request.client_ip, BLOCK_REG, 'jo') then
        return {status = 403}
    end

    -- 其他继续回源
    return event.request
end

return _M

File: Custom_responses_that_don’t_hit_origin_servers/python/prevent_a_specific_ip.py
import re

BLOCK_REG = '1\.2\.3\.\d+'


async def handler(event):
    request = event.request

    # 如果 client ip 属于 1.2.3.0/24
    # 返回 403
    if re.match(BLOCK_REG, request.client_ip):
        return {'status': 403}

    # 其他继续回源
    return Request(request=request)
